### ISSUE TITLE
Windows dependency links are wrong

### ISSUE THREAD

**Issue by TheB3arGod** [fireice-uk/xmr-stak#2408]
`...` are the placeholder for your answers. Please answer each question!

**Describe the bug**
The OpenSSL/Hwloc and Microhttpd links for CUDA 8 and CUDA 9 / AMD / CPU are swapped.
Using the CUDA 8 dependency link (*/v1/*) compiles properly using the command `cmake … -T v141,host=x64`.
Using the CUDA 9 dependency link (*/v2/*) throws a link error: `LINK : fatal error C1900: Il mismatch between 'P1' version '20161212' and 'P2' version '20150812'`

**Which operating system do you use? **

```
Windows 10
```

**To Reproduce**
```
# Please post all commands and the output.

Download both OpenSSL/Hwloc and Microhttpd dependency zip files
Extract CUDA 9 xmr-stak-dep.zip to C:\xmr-stak-dep
cmake -G "Visual Studio 15 2017 Win64" -T v141,host=x64 -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=ON -DCPU_ENABLE=OFF ..
cmake --build . --config Release --target install
ERROR!


Extract CUDA 8 xmr-stak-dep.zip to C:\xmr-stak-dep
cmake -G "Visual Studio 15 2017 Win64" -T v141,host=x64 -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=ON -DCPU_ENABLE=OFF ..
cmake --build . --config Release --target install
Success!
```

**Additional information.**

```
# run `cmake -LA .` in the build folder and add the output here
NA
```

**Feel free to add more information.**
```
NA
```

**Comment by psychocrypt** [comment]
for v1 dependencies you must use v140 NOT v141. Thats the reason why it is
not working.

Jonathan Lin <notifications@github.com> schrieb am Mi., 17. Apr. 2019,
00:50:

**Comment by TheB3arGod** [comment]
No, I don't think you read this correctly.  v1 DID work with v141.  I think you mixed up your links.

**Comment by Ryo-RU** [comment]
overall, if you download VS now (16 2019) all is fine with 142 and links from this guide: https://www.reddit.com/r/XmrStak/wiki/guides/startup#wiki_3._windows_7.2F10_.2B_nvidia_.28compiling_from_source.29
(tested on GTX 1060 + GT640)

**Comment by psychocrypt** [comment]
i will check the links again.

**Comment by Spudz76** [comment]
Check your source md5sums because they still match what files I had from long, long, long ago...
```
5869e0f5a670fe62059d8bc523902db6 *xmr-stak-dep_v1.zip
b143b0fdf8130592f248651c2519f841 *xmr-stak-dep_v2.zip
aa339e761a22e3b92e9a05316bfe3369 *xmr-stak-dep_v3.zip
```

**Comment by Spudz76** [comment]
Also re-unpacked them and built CUDA8 just fine no link errors

Did you set any `CMAKE_PREFIX_PATH`?
Did you destroy the whole `C:\xmr-stak-dep\` path before the next unpack?

I keep mine in `<xmr-stak>/dep/v1/` or `/v2/` or `/v3/` and set prefix to `CMAKE_PREFIX_PATH=..\dep\v1\hwloc;..\dep\v1\libmicrohttpd;..\dep\v1\openssl` or the other versions, as appropriate to match compiler (which is dictated by CUDA requirements, per version)

You also have to force the correct compiler before calling CMake regardless which toolkit you tell it.
